pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -46,11 +46,6 @@ func (e *ErrorWithCodes) Codes() []gardencorev1beta1.ErrorCode {
 	return e.codes
 }
 
-// Unwrap rettieves the error from ErrorWithCodes.
-func (e *ErrorWithCodes) Unwrap() error {
```

Comment:
We are you removing this function instead of removing the call to `utilerrors.Unwrap` in `NewWrappedLastErrors`?
I don't think, there is anything wrong with this `Unwrap` implementation but rather with the `utilerrors.Unwrap` call.

Reply:
I tried with removing the utilerrors.Unwrap` call, but it didn't work. Only removing this `Unwrap` implementation yielded the expected result. To be honest, I find this entire error wrapping and `Causer` interfaces highly confusing. I don't know what's wrong or where it fails.

Reply:
Can you try implementing `Unwrap` on `reconciliationError`?
I think, the problem that you encountered was that something in the error chain doesn't implement `Unwrap` and thus `DeprecatedDetermineErrorCodes` can't find the respective `Coder` via `errors.As`.

`reconciliationError` is constructed via `utilerrors.WithID` here: https://github.com/gardener/gardener/blob/f9abbec8409b02ca1f02fececf1bbb7ad05f7969/pkg/utils/flow/flow.go#L291

Reply:
Thanks for the hint, this helped indeed! :)